src/fsharp/CompileOps.fs

```diff
@@ -1835,6 +1835,67 @@ let ComputeMakePathAbsolute implicitIncludeDir (path: string) =
     with 
         :? System.ArgumentException -> path  
 
+//
+// Add your features to this List - in code use languageVersion.SupportsFeature(LanguageFeatures.yourFeature) 
+// a return value of false means your feature is not supported by the users language selection
+// All new language features added from now on must be protected by this.
+// Note:
+//   *  The fslang design process will require a decision about feature name and whether it is required.
+//   *  When a feature is assigned a release language, we will scrub the code of feature references and apply
+//      the Release Language version.
+//
+[<RequireQualifiedAccess>]
+type LanguageFeature =
+    | LanguageVersion47 = 0
+    | Nullness = 1000
+    | ScriptingPackageManagement = 1001
+
+type LanguageVersion (specifiedVersion) =
+    static let previewVersion = 4.8m                // Language version when preview specified
+    static let defaultVersion = 4.7m                // Language version when default specified
+    static let latestVersion = defaultVersion       // Language version when latest specified
+    static let latestMajorVersion = 4.7m            // Language version when latestmajor specified
+
+    static let validOptions = [| "preview"; "default"; "latest"; "latestmajor" |]
```

Comment:
I'm actually not so hot on the idea of an explicit value called `default`. C# has it, but nobody uses it because it's the implicit value, and the meaning of that value is now also going to be influenced by TFM moving forward.

Reply:
Is 4.6 going to be a valid option, or are we starting it at 4.7?

Reply:
We will start from 4.7, unless someone can do the work to use this in the existing PR's we are adding to 4.7.  If we do that work then it will be 4.6.

Reply:
I'm going to push back (gently) on allowing `default` to be specified; it can make certain testing easier.  E.g., in the future we may do something like this:

``` F#
for langVer in ["default"; "4.7"; "4.8"] do
    exec <| sprintf "fsc.exe --langVersion:%s MyTestFile.fs" langVer
```

Reply:
C# is the model, and it has default.  I intend to keep it, because if any one learned it for C# then they can use it for F#.

Reply:
Just to be clear, _practically nobody_ uses `default` as a value in C#-land. We even ignored it by accident in the 16.0 language version work and that had zero negative effects.

Reply:
@cartermp agreed, it is a pointless design.  latest and default will always be the same value in any rational world, and latest is more memorable.  However, it costs us almost nothing, and saves us from having to deal with the distinction without a difference.